### PR TITLE
Add findLast and findLastIndex functions

### PR DESCRIPTION
**Pull request by LandonSchropp**
I tackled a few things in this pull request:

- Implement findLast
- Implement findLastIndex
- Fix findIndex signature

I used a for loop to try and maximize performance. 🙂

I was running into trouble with the lazy logic and the iteration order, so I ended up cutting it. To be honest, I was having a little trouble following what the intent was with it. Please let me know if you'd like me to add it back in, and if you have any suggestions on how to tackle it.

- [x] Typedoc added for new methods and updated for changed
- [x] Tests added for new methods and updated for changed
- [x] New methods added to `src/index.ts`
- [x] New methods added to `mapping.md`

### CODE REVIEW

**Comment by LandonSchropp** [comment]
As an alternative, this is much more concise, but I felt like the utility library should strive for best performance.

```
createPipe(reverse, find(predicate))
```

File: src/findIndex.ts [hunk 1]
```diff
@@ -85,10 +85,10 @@ export namespace findIndex {
   export function indexed<T>(
     array: readonly T[],
     fn: PredIndexed<T, boolean>
-  ): T | undefined;
+  ): number;
   export function indexed<T>(
     fn: PredIndexed<T, boolean>
-  ): (array: readonly T[]) => T | undefined;
+  ): (array: readonly T[]) => number;
```
Comment (LandonSchropp) [inline src/findIndex.ts:91]: I found this while looking through the `findIndex` implementation, and it seemed like a bug. 🙂

Comment (TkDodo) [reply src/findIndex.ts:91]: yes, thank you. probably a copy-paste error (looks like the `find` interface). `findIndex` returns a `number` 😅

**Review by TkDodo — APPROVED** [review]

**Comment by TkDodo** [comment]
please check the prettier formatting - looks good otherwise ❤️

**Comment by LandonSchropp** [comment]
@TkDodo Sorry for the delay! I had a baby. 🐣 

Should be ready to go now. Let me know if you need anything else for me! Thanks for taking a look at this.

**Comment by LandonSchropp** [comment]
@TkDodo Hey, sorry to bug you, but have you had a chance to look at my two PRs? I'd love to get these merged if possible so I can use them in my project. 🙂

**Comment by TkDodo** [comment]
sorry for the delay - I'll try to take a look at all open PRs today

**Review by TkDodo — APPROVED** [review]

**Comment by TkDodo** [comment]
> I had a baby. 🐣

❤️ congratulations. Thank you for the PR 🚀

**Comment by LandonSchropp** [comment]
Thank you!

**Pull request merged**